enterprise_access/apps/provisioning/api.py

```diff
@@ -0,0 +1,64 @@
+"""
+Python API for provisioning operations.
+"""
+import logging
+
+from ..api_client.lms_client import LmsApiClient
+
+logger = logging.getLogger(__name__)
+
+
+def get_or_create_enterprise_customer(*, name, slug, country, **kwargs):
+    """
+    Get or creates an enterprise customer with the provided arguments.
+    """
+    client = LmsApiClient()
+    existing_customer = client.get_enterprise_customer_data(enterprise_customer_slug=slug)
+    if existing_customer:
+        logger.info('Provisioning: enterprise_customer slug %s already exists', slug)
+        return existing_customer
```

Comment:
[non-blocking idea] Maybe log a warning if the data on the found customer does not match the request data?

Reply:
oh, that's a good idea, lemme noodle on that